Watchman.DomainModel/Responses/ResponsesMessageService.cs

```diff
@@ -0,0 +1,185 @@
+                _messagesService.SendEmbedMessage("Nadpisane responses:", description: null, message);
+            }
+        }
+
+        private void PrintAllResponses()
+        {
+            var messagesAboutAllResponses = GetMessagesAboutAllResponses();
+            for (int i = 0; i < messagesAboutAllResponses.Count; ++i)
+            {
+                if (i == 0)
+                {
+                    _messagesService.SendEmbedMessage("Wszystkie responses:", description: null, messagesAboutAllResponses[0]);
+                }
+
+                else
+                {
+                    _messagesService.SendEmbedMessage(title: null, description: null, messagesAboutAllResponses[i]);
+                }
+            }
+        }
+
+        private List<List<KeyValuePair<string, string>>> GetMessagesAboutDefaultResponses()
+        {
+            var defaultResponses = _responsesDatabase.GetResponsesFromBase()
+                                   .Where(x => x.IsDefault)
+                                   .ToList();
+            var messagesAboutDefaultResponses = new List<List<KeyValuePair<string, string>>>();
+
+            for (int i = 0; i < defaultResponses.Count; ++i)
+            {
+                if (i % MAX_NUMBER_OF_MESSAGE_FIELDS == 0)
+                {
+                    messagesAboutDefaultResponses.Add(new List<KeyValuePair<string, string>>(MAX_NUMBER_OF_MESSAGE_FIELDS));
+                }
+
+                var onEvent = defaultResponses[i].OnEvent;
+                var message = GetMessageWithoutFramesAndBold(defaultResponses[i].Message);
+                var oneField = new KeyValuePair<string, string>(onEvent, message);
+                messagesAboutDefaultResponses.Last().Add(oneField);
+            }
+
+            return messagesAboutDefaultResponses;
+        }
+
+        private List<List<KeyValuePair<string, string>>> GetMessagesAboutCustomResponses()
+        {
+            var customResponses = _responsesDatabase.GetResponsesFromBase()
+                                  .Where(x => !x.IsDefault)
+                                  .ToList();
+            var messagesAboutCustomResponses = new List<List<KeyValuePair<string, string>>>();
+
+            for (int i = 0; i < customResponses.Count; ++i)
+            {
+                if (i % MAX_NUMBER_OF_MESSAGE_FIELDS == 0)
+                {
+                    messagesAboutCustomResponses.Add(new List<KeyValuePair<string, string>>(MAX_NUMBER_OF_MESSAGE_FIELDS));
+                }
+
+                var onEvent = customResponses[i].OnEvent;
+                var message = GetMessageWithoutFramesAndBold(customResponses[i].Message);
+                var oneField = new KeyValuePair<string, string>(onEvent, message);
+                messagesAboutCustomResponses.Last().Add(oneField);
+            }
+
+            return messagesAboutCustomResponses;
+        }
+
+        private List<List<KeyValuePair<string, string>>> GetMessagesAboutAllResponses()
```

Comment:
W sumie to 3 razy ten sam kod powtarzasz
Zrób jedną metodę od parsowania responses na wiadomości
Potem w każdych metodach (GetMessagesAbout...) będziesz pobierać z bazy odpowiednie responses, parsować tamtą metodą i zwracać wynik

Reply:
1. chodzi ci, że powtarzam kod w metodach GetMessagesAbout...? Jeśli tak mam to zrobić, że tą pętle dać do innej metody, a metody GetMessagesAbout... po prostu by zwracały odpowiednie listy responses?
2. Może zrobię też jakoś metodę ogólną dla metod Print...? tam też powtarza się kod

Reply:
Jeśli zrobisz to co teraz napisałem w poprzednim komentarzu to te metody będzie można bardzo uprościć albo będą już wgl nie potrzebne